apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +107,175 @@ export class ChangeAction extends BaseRushAction {
+        this._overwriteFlagParameter
+      ]).map((parameter) => {
+        return parameter.value
+        ? (
+          `The {${this._bulkChangeParameter.longName} parameter cannot be provided with the ` +
+            `${this._verifyParameter.longName} parameter`
+          )
+        : '';
+      }).filter((error) => error !== '');
+      if (errors.length > 0) {
+        errors.forEach((error) => console.error(error));
+        throw new AlreadyReportedError();
+      }
+
       this._verify();
-      return Promise.resolve();
+      return;
     }
 
-    this._sortedProjectList = this._getChangedPackageNames().sort();
-    if (this._sortedProjectList.length === 0) {
+    const sortedProjectList: string[] = this._getChangedPackageNames().sort();
+    if (sortedProjectList.length === 0) {
       this._logNoChangeFileRequired();
       this._warnUncommittedChanges();
-      return Promise.resolve();
+      return;
     }
 
-    this._prompt = inquirer.createPromptModule();
-    this._changeFileData = new Map<string, IChangeFile>();
-    this._changeComments = ChangeFiles.getChangeComments(this._getChangeFiles());
+    this._warnUncommittedChanges();
+
+    const promptModule: inquirer.PromptModule = inquirer.createPromptModule();
+    let changeFileData: Map<string, IChangeFile> = new Map<string, IChangeFile>();
+    let interactiveMode: boolean = false;
+    if (this._bulkChangeParameter.value) {
+      if (
+        !this._bulkChangeBumpTypeParameter.value ||
+        (
+          !this._bulkChangeMessageParameter.value &&
+          this._bulkChangeBumpTypeParameter.value !== ChangeType[ChangeType.none]
+        )
+      ) {
+        throw new Error(
+          `The ${this._bulkChangeBumpTypeParameter.longName} and ${this._bulkChangeMessageParameter.longName} ` +
+          `parameters must provided if the ${this._bulkChangeParameter.longName} flag is provided. If the value ` +
+          `"${ChangeType[ChangeType.none]}" is provided to the ${this._bulkChangeBumpTypeParameter.longName} ` +
+          `parameter, the ${this._bulkChangeMessageParameter.longName} parameter may be omitted.`
+        );
+      }
 
-    return this._promptLoop()
-      .catch((error: Error) => {
-        throw new Error(`There was an error creating a change file: ${error.toString()}`);
+      const email: string | undefined = this._changeEmailParameter.value || this._detectEmail();
+      if (!email) {
+        throw new Error(
+          'Unable to detect Git email and an email address wasn\'t provided using the ' +
+          `${this._changeEmailParameter.longName} paramter.`
+        );
+      }
+
+      const errors: string[] = [];
+
+      const comment: string = this._bulkChangeMessageParameter.value || '';
+      const changeType: string = this._bulkChangeBumpTypeParameter.value;
+      for (const packageName of sortedProjectList) {
+        const allowedBumpTypes: string[] = Object.keys(this._getBumpOptions(packageName));
```

Comment:
Please test these two cases:

- When a `lockStepVersion` version policy is in use, the bump type is determined by **version-policies.json**.  The Rush packages are configured [in this way](https://github.com/microsoft/rushstack/blob/b1aa16fc0ea6aa5152857b187523c34b28e2d4a0/common/config/rush/version-policies.json#L82).  I believe the change files get created with `type=none` (?), although that design should probably be improved.
- The `lockedMajor` setting in **version-policies.json** can be used to allow only minor/patch choices.  Please test that `--bump-type` cannot be used to bypass this.

Reply:
The former works correctly, the latter creates a `"major"` changefile.

I'll fix the issue with the latter.

Reply:
Fixed in acd599d934a45db1b58cece773ab97d3ef412098